cmake/modules/FindQatDrv.cmake

```diff
@@ -0,0 +1,122 @@
+	    "Remember to set an environment variable for the package root directory, "
+	    "i.e. \"export ICP_ROOT=/opt/APP/driver/QAT\".")
+  endif()
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_TYPE=QAT[0-9]+.[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_TYPE=QAT([0-9]+.[0-9]+)$"
+         "\\1" QAT_VERSION1 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_OS=[A-Z]$")
+  string(REGEX REPLACE "^PACKAGE_OS=([A-Z])$"
+         "\\1" QAT_VERSION2 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_MAJOR_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_MAJOR_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION3 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_MINOR_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_MINOR_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION4 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_PATCH_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_PATCH_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION5 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_BUILD_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_BUILD_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION6 "${QAT_VERSION_LINE}")
+  set(version_a "v${QAT_VERSION1}.${QAT_VERSION2}.${QAT_VERSION3}.${QAT_VERSION4}.${QAT_VERSION5}-${QAT_VERSION6}")
+  set(version "${QAT_VERSION3}.${QAT_VERSION4}.${QAT_VERSION5}")
+  if(version VERSION_LESS EXPECTED_version)
+    message(FATAL_ERROR "QAT ${version_a} in ${source_dir} is not new enough. "
+	    "Please either \"rm -rf ${source_dir}\" "
+            "so you can download and install QAT driver ${expected_version} manually at the link "
+	    "${expected_version_url}, or make sure ${source_dir} "
+            "contains a full copy of QAT driver ${expected_version}. "
+            "Remember to set an environment variable for the package root directory, "
+            "i.e. \"export ICP_ROOT=/opt/APP/driver/QAT\"." )
+  else()
+    message(STATUS "QAT driver (${version} >= ${EXPECTED_version}) already in ${source_dir}")
+  endif()
+endfunction()
+	
+find_qatdrv_source_dir()
+check_qatdrv_version(${expected_version} ${expected_version_url})  
+
+set(QatDrv_INCLUDE_DIRS
+    $ENV{ICP_ROOT}/quickassist/include
+    $ENV{ICP_ROOT}/quickassist/include/dc
+    $ENV{ICP_ROOT}/quickassist/lookaside/access_layer/include
+    $ENV{ICP_ROOT}/quickassist/include/lac
+    $ENV{ICP_ROOT}/quickassist/utilities/libusdm_drv
+    $ENV{ICP_ROOT}/quickassist/utilities/libusdm_drv/include
+   )
+set(source_dir $ENV{ICP_ROOT})
+set(QatDrv_COMPONENTS qat_s usdm_drv_s)
```

Comment:
i'd suggest just use `QatDrv_FIND_COMPONENTS`. and pass `qat_s` and `usdm_drv_s` to `find_package(QatDrv)` as the argument for the `COMPONENTS` option.